### PR TITLE
User.is_anonymous is not callable in Django 2.x.

### PR DESCRIPTION
**Pull request by lamby**

### CODE REVIEW

**Comment by moggers87** [comment]
This breaks on Django 1.8 (which is still supported upstream until April 2018 iirc)

**Comment by lamby** [comment]
@moggers87 Good shout. Updated patch.

**Comment by lamby** [comment]
@Bouke D'oh. Updated and rebased against master.... :)

File: two_factor/utils.py [hunk 1]
```diff
@@ -61,3 +63,14 @@ def totp_digits():
     for totp tokens. Defaults to 6
     """
     return getattr(settings, 'TWO_FACTOR_TOTP_DIGITS', 6)
+
+
+def user_is_anonymous(user):
+    if not user:
+        return False
```
Comment (Bouke) [inline two_factor/utils.py]: Should return `True` here.

**Comment by Bouke** [comment]
The linter is blocking the PR because of the following error;

```
ERROR: /home/travis/build/Bouke/django-two-factor-auth/two_factor/utils.py Imports are incorrectly sorted.
--- /home/travis/build/Bouke/django-two-factor-auth/two_factor/utils.py:before	2017-11-05 09:08:47.232806
+++ /home/travis/build/Bouke/django-two-factor-auth/two_factor/utils.py:after	2017-11-05 09:08:56.383716
@@ -1,5 +1,4 @@
 import django
-
 from django.conf import settings
 from django_otp import devices_for_user
```

Once this is resolved, the PR can be merged.

**Comment by lamby** [comment]
@Bouke I thiiink that's it.. :)

File: two_factor/utils.py [hunk 2]
```diff
@@ -61,3 +62,14 @@ def totp_digits():
...
     """
     return getattr(settings, 'TWO_FACTOR_TOTP_DIGITS', 6)
+
+
+def user_is_anonymous(user):
+    if not user:
+        return True
+
+    # https://docs.djangoproject.com/en/dev/releases/1.10/#using-user-is-authenticated-and-user-is-anonymous-as-methods
+    if django.VERSION < (2, 0):
```
Comment (richardowen) [inline two_factor/utils.py]: Could this be changed to `< (1, 10)`? That will prevent DeprecationWarnings being emitted on Django 1.10 and 1.11.

**Comment by lamby** [comment]
@Bouke @richardowen Re-pushed with requested change & matching commit message.

**Comment by Bouke** [comment]
Thank you for contributing!

**Comment by lamby** [comment]
Thanks @Bouke  :)

**Pull request merged**